packages/dev/core/src/XR/features/WebXRDepthSensing.ts

```diff
@@ -352,6 +657,33 @@ export class WebXRDepthSensing extends WebXRAbstractFeature {
             }
         });
     }
+
+    private _getInternalTextureFromDepthInfo(): InternalTexture {
+        const engine = this._xrSessionManager.scene.getEngine();
+        const dataFormat = this.depthDataFormat;
+        const textureType = this._textureType;
+        if (!this._width || !this._height || !this._cachedWebGLTexture) {
+            throw new Error("Depth information is not available");
+        }
+        const internalTexture = new InternalTexture(engine, InternalTextureSource.Unknown, true);

```

Comment:
could we use engine.wrapWebGLTexture for the common part ?

Reply:
sure